src/sentry/static/sentry/app/views/alerts/rules/details/metricChart.tsx

```diff
@@ -1,145 +1,250 @@
 import React from 'react';
+import color from 'color';
 import moment from 'moment';
 
+import Graphic from 'app/components/charts/components/graphic';
 import MarkLine from 'app/components/charts/components/markLine';
 import LineChart, {LineChartSeries} from 'app/components/charts/lineChart';
 import space from 'app/styles/space';
-import {Series} from 'app/types/echarts';
+import {ReactEchartsRef, Series} from 'app/types/echarts';
 import theme from 'app/utils/theme';
-import {Trigger} from 'app/views/settings/incidentRules/types';
+import {IncidentRule, Trigger} from 'app/views/settings/incidentRules/types';
 
 import {Incident} from '../../types';
 
+const X_AXIS_BOUNDARY_GAP = 15;
+
 type Props = {
   data: Series[];
+  rule?: IncidentRule;
   incidents?: Incident[];
   warningTrigger?: Trigger;
   criticalTrigger?: Trigger;
 };
 
-function createThresholdSeries(color: string, threshold: number): LineChartSeries {
+type State = {
+  width: number;
+  height: number;
+};
+
+function createThresholdSeries(lineColor: string, threshold: number): LineChartSeries {
   const criticalThresholdLine = {
     seriesName: 'Threshold Line',
     type: 'line',
     markLine: MarkLine({
       silent: true,
-      lineStyle: {color, type: 'dashed', width: 1},
+      lineStyle: {color: lineColor, type: 'dashed', width: 1},
```

Comment:
was this changed to make it look more consistent?

Reply:
I was getting a TS warning, because color was declared in the upper context with `import color from 'color';`

Reply:
ahh didn't catch that 
makes sense 👍